R/qenv-eval_code.R

```diff
@@ -21,31 +21,33 @@ setGeneric("eval_code", function(object, code) standardGeneric("eval_code"))
 
 #' @rdname eval_code
 #' @export
-setMethod("eval_code", signature = c("qenv", "expression"), function(object, code) {
-  id <- sample.int(.Machine$integer.max, size = length(code))
+setMethod("eval_code", signature = c("qenv", "character"), function(object, code) {
+  id <- sample.int(.Machine$integer.max, size = 1)
 
   object@id <- c(object@id, id)
   object@env <- rlang::env_clone(object@env, parent = parent.env(.GlobalEnv))
+  code <- paste(code, collapse = "\n")
   object@code <- c(object@code, code)
```

Comment:
1. Does `code` accept character vectors of any length?
2. Does `@code` store a 1-per-added character vector or a single string

Reply:
@chlebowa `code` is accepted as character of any length and it gets pasted. then `@code` is extended by length 1, each time an `eval_code` is run. And that's what I think we agreed, that each code input is pasted and stored as one string, but each eval_code execution aggregates `object@code <- c(object@code, code)`

Reply:
I thought we would collapse in `@code` as well.

Reply:
no we don't nor we collapse warnings nor errors. the lenght of id, code, warning, error increases of 1 each time the eval_code or new_qenv is run